webapp/channels/src/components/search_results/index.tsx

```diff
@@ -80,8 +81,15 @@ function makeMapStateToProps() {
         const currentSearch = (getCurrentSearchForSearchTeam(state) as unknown as Record<string, any>) || {};
         const currentTeamName = getCurrentTeam(state)?.name ?? '';
 
+        if (ownProps.isPinnedPosts) {
+            results = results.sort((postA: Post|FileSearchResultItem, postB: Post|FileSearchResultItem) => postB.create_at - postA.create_at);
```

Comment:
This needs to be moved into `getSearchResults` or the `// Cache posts and channels` block above to avoid returning a new `results` every time Redux updates. This component is a bit odd because the `mapStateToProps` does some of that memoization itself instead of doing it in `getSearchResults`.

Could you move this sorting into the block above so that we only sort this once when `results` changes?